File: PGL/React/practica_examen_app_peliculas/src/models/Pelicula.ts
import { Categoria } from "./Categoria";

class Pelicula {
  private id: string;
  private titulo: string;
  private direccion: string;
  private actores: string;
  private argumento: string;
  private imagen: string;
  private trailer: string;
  private categoria: Categoria[];

  constructor(
    id: string,
    titulo: string,
    direccion: string,
    actores: string,
    argumento: string,
    imagen: string,
    trailer: string,
    categoria: Categoria[]
  ) {
    this.id = id;
    this.titulo = titulo;
    this.direccion = direccion;
    this.actores = actores;
    this.argumento = argumento;
    this.imagen = imagen;
    this.trailer = trailer;
    this.categoria = categoria;
  }

  getId = (): string => {
    return this.id;
  }

  setId(id: string) {
    this.id = id;
  }

  getTitulo(): string {
    return this.titulo;
  }

  setTitulo(titulo: string) {
    this.titulo = titulo;
  }

  getDireccion(): string {
    return this.direccion;
  }

  setDireccion(direccion: string) {
    this.direccion = direccion;
  }

  getActores(): string {
    return this.actores;
  }

  setActores(actores: string) {
    this.actores = actores;
  }

  getArgumento(): string {
    return this.argumento;
  }

  setArgumento(argumento: string) {
    this.argumento = argumento;
  }

  getImagen(): string {
    return this.imagen;
  }

  seImagen(imagen: string) {
    this.imagen = imagen;
  }

  getTrailer(): string {
    return this.trailer;
  }

  setTrailer(trailer: string) {
    this.trailer = trailer;
  }

  getCategoria(): Categoria[] {
    return this.categoria;
  }

  setCategoria(categoria: Categoria[]) {
    this.categoria = categoria;
  }

  getCategoriasComoString(): string {
    //const categoriasStringArray = (this.getCategoria() as Categoria[]).map(c => c.getNombre());
    const categoriasStringArray = this.getCategoria().map(c => c.nombre);

    // // Utiliza join para unir los nombres de categorías en una cadena
    const categoriasComoString = categoriasStringArray.join(', ');

    return categoriasComoString;
  }
}

export { Pelicula }

File: AED/Springboot/Peliculas/src/main/java/es/iespuertodelacruz/sgp/peliculas/entities/Pelicula.java
package es.iespuertodelacruz.sgp.peliculas.entities;

import java.io.Serializable;
import jakarta.persistence.*;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;


/**
 * The persistent class for the Peliculas database table.
 * 
 */
@Entity
@Table(name="Peliculas")
@NamedQuery(name="Pelicula.findAll", query="SELECT p FROM Pelicula p")
public class Pelicula implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	private int id;

	private String actores;

	private String argumento;

	private String direccion;

	private String imagen;

	private String titulo;

	private String trailer;

	//bi-directional many-to-many association to Categoria
	@JsonIgnore
	@ManyToMany(fetch= FetchType.LAZY)
	@JoinTable(
		name="PeliculaCategoria"
		, joinColumns={
			@JoinColumn(name="id")
			}
		, inverseJoinColumns={
			@JoinColumn(name="categoria_id")
			}
		)
	private List<Categoria> categorias;

	public Pelicula() {
	}

	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getActores() {
		return this.actores;
	}

	public void setActores(String actores) {
		this.actores = actores;
	}

	public String getArgumento() {
		return this.argumento;
	}

	public void setArgumento(String argumento) {
		this.argumento = argumento;
	}

	public String getDireccion() {
		return this.direccion;
	}

	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}

	public String getImagen() {
		return this.imagen;
	}

	public void setImagen(String imagen) {
		this.imagen = imagen;
	}

	public String getTitulo() {
		return this.titulo;
	}

	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}

	public String getTrailer() {
		return this.trailer;
	}

	public void setTrailer(String trailer) {
		this.trailer = trailer;
	}

	public List<Categoria> getCategorias() {
		return this.categorias;
	}

	public void setCategorias(List<Categoria> categorias) {
		this.categorias = categorias;
	}

}